pytest-embedded/pytest_embedded/plugin.py

```diff
@@ -1252,7 +1253,35 @@ def _duplicate_items(items: t.List[_T]) -> t.List[_T]:
         return duplicates
 
     @pytest.hookimpl(hookwrapper=True, trylast=True)
-    def pytest_collection_modifyitems(self, items: t.List[Function]):
+    def pytest_collection_modifyitems(self, config: Config, items: t.List[Function]):
+        for item in items:
+            skip_marker = item.get_closest_marker('skip_if_soc')
+            if not skip_marker:
+                continue
+
+            target = config.getoption('--target', None)
+            if hasattr(item, 'callspec'):
+                target = item.callspec.params.get('target', None)
+            if target == 'auto' or target is None:
+                logging.warning(
+                    f"Test item '{item.originalname}': "
+                    "Ensure 'target' is included in the test @pytest.mark.parametrize when using 'skip_if_soc', "
+                    "or provide the --target argument when running tests (except 'auto' and multi-dut)."
+                )
+                continue
+            if '|' in target:
```

Comment:
There is a question about multi-DUT support: should we support it? If yes, I think we need to do something like masking -> `esp32|esp32c5`, where the mask would be `0|1`, or for `esp32c5|esp32c6`, use the default mask(by default check all), or for `esp32c5|esp32`, mask it as`1|0`.

ex:
```python
@pytest.mark.skip_if_soc("SOC_BLE_SUPPORTED == 1", target_check_mask='1|0')

```

Reply:
nice catch. how about an api like `..., check_on_duts=[0, 1]` or `check_on_duts='all'` (default)?

Reply:
I think such case is rare enough that we can simply move the "skip" check into the test case itself. And then we don't have to introduce or handle an additional argument which 99% of the time won't be used.

Reply:
Ok, so let's not add this overwhelm here; just raise a warning if there few dut.